ArcGISRuntimeSDKQt_CppSamples/Analysis/FindFeaturesUtilityNetwork/FindFeaturesUtilityNetwork.cpp

```diff
@@ -0,0 +1,314 @@
+
+  m_deviceFeatureTable = new ServiceFeatureTable(QUrl("https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer/100"), this);
+  m_deviceLayer = new FeatureLayer(m_deviceFeatureTable, this);
+  connect(m_deviceLayer, &FeatureLayer::selectFeaturesCompleted, this, [this](QUuid)
+  {
+    busy = false;
+    emit busyChanged();
+  });
+
+  m_lineFeatureTable = new ServiceFeatureTable(QUrl("https://sampleserver7.arcgisonline.com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer/115"), this);
+  m_lineLayer = new FeatureLayer(m_lineFeatureTable, this);
+  m_lineLayer->setRenderer(new SimpleRenderer(m_lineSymbol, this));
+
+  // Add electric distribution lines and electric devices layers
+  m_map->operationalLayers()->append(m_lineLayer);
+  m_map->operationalLayers()->append(m_deviceLayer);
+
+  connect(m_map, &Map::doneLoading, this, [this](Error e)
+  {
+    if (!e.isEmpty())
+    {
+      dialogText = QString(e.message() + " - " + e.additionalMessage());
+      emit dialogVisibleChanged();
+      return;
+    }
+
+    m_graphicsOverlay = new GraphicsOverlay(this);
+    m_mapView->graphicsOverlays()->append(m_graphicsOverlay);
+
+    m_utilityNetwork = new UtilityNetwork(m_serviceUrl, m_map, this);
+    connect(m_utilityNetwork, &UtilityNetwork::traceCompleted, this, [this](QUuid)
+    {
+      dialogVisible = true;
+      emit dialogVisibleChanged();
+
+      if (m_utilityNetwork->traceResult()->isEmpty())
+      {
+        busy = false;
+        dialogText = QString("Trace complete with no results.");
+        emit dialogTextChanged();
+        emit busyChanged();
+        return;
+      }
+
+      dialogText = QString("Trace completed.");
+      emit dialogTextChanged();
+
+      UtilityTraceResult* result = m_utilityNetwork->traceResult()->at(0);
+
+      if (!result)
+        return;
+
+      const QList<UtilityElement*> elements = static_cast<UtilityElementTraceResult*>(result)->elements(this);
+
+      QueryParameters deviceParams;
+      QueryParameters lineParams;
+      QList<qint64> deviceObjIds;
+      QList<qint64> lineObjIds;
+
+      for (UtilityElement* item : elements)
+      {
+        if (item->networkSource()->name() == "Electric Distribution Device")
+        {
+          deviceObjIds.append(item->objectId());
+        }
+        else if (item->networkSource()->name() == "Electric Distribution Line")
+        {
+          lineObjIds.append(item->objectId());
+        }
+      }
+
+      deviceParams.setObjectIds(deviceObjIds);
+      lineParams.setObjectIds(lineObjIds);
+
+      m_deviceLayer->selectFeatures(deviceParams, SelectionMode::Add);
+      m_lineLayer->selectFeatures(lineParams, SelectionMode::Add);
+    });
+
+    m_utilityNetwork->load();
+    m_traceParams = new UtilityTraceParameters(UtilityTraceType::Connected, {}, this);
+
+    connectSignals();
```

Comment:
Re-worked the logic around this a little bit to ensure the map has been loaded as well as the utility network before connecting those signals to avoid dereferencing a null object or one that is not loaded.